in_toto/models/link.py

```diff
@@ -141,3 +141,59 @@ def __init__(self, **kwargs):
     self.byproducts = kwargs.get("byproducts", {})
     self.command = kwargs.get("command", [])
     self.return_value = kwargs.get("return_value", None)
+
+    self.validate()
+
+  def _validate_type(self):
+    """Private method to check that `_type` is set to "link"."""
+    if self._type != "link":
+      raise securesystemslib.exceptions.FormatError(
+          "Invalid Link: field `_type` must be set to 'link', got: {}"
+          .format(self._type))
+
+
+  def _validate_materials(self):
+    """Private method to check that `materials` is a `dict` of `HASHDICTs`."""
+    if not isinstance(self.materials, dict):
+      raise securesystemslib.exceptions.FormatError(
+          "Invalid Link: field `materials` must be of type list, got: {}"
```

Comment:
I wonder if we can coalesce this to a format string (probably with our own exception class) and just do:
```
raise InvalidLinkException(field='materials', got=type(self.materials))
```

Here and elsewhere.

Reply:
We could and should, but let's do this elsewhere. Care for a "revisit exception taxonomy" ticket?

Reply:
FYI: https://github.com/in-toto/in-toto/issues/126